src/Orleans.Transactions/State/TransactionParticipantExtension.cs

```diff
@@ -3,11 +3,14 @@
 using System;
 using System.Collections.Generic;
 using System.Threading.Tasks;
+using Newtonsoft.Json;
 
 namespace Orleans.Transactions
 {
+    [Serializable]
     public class TransactionParticipantExtension : ITransactionParticipantExtension
```

Comment:
y does this need to be serializable?

Reply:
it is one of the fields of TransactionParticipantExtensionWrapper, which needs to be serialized right?

Reply:
To my knowledge it shouldn't be.   The ITransactionParticipantExtension in the wrapper should be a grain cast to the extension with this concrete class being what handles the calls, not this class itself.